pkg/clients/openshift.go

```diff
@@ -124,14 +126,34 @@ func NewOpenShiftClient() (OpenShiftClient, error) {
 }
 
 func newKubeClient(token string) *kubernetes.Clientset {
-	config, err := rest.InClusterConfig()
+	config, err := getConfig()
 	if token != "" {
 		config.BearerToken = token
 	}
-	log.Tracef("Creating new OpenShift client with: %+v", config)
+	log.Tracef("Creating new OpenShift client %v", config.Host)
 	clientset, err := kubernetes.NewForConfig(config)
 	if err != nil {
 		panic(err.Error())
 	}
 	return clientset
 }
+
+func getConfig() (*rest.Config, error) {
```

Comment:
Will this function, via either `rest.InClusterConfig` or `clientcmd.BuildConfigFromFlags`, allow us to specify `KUBECONFIG` env. var. to provide the cluster config and credentials?  That is, how can we use `getConfig` to allow us to specify the sa ca.crt and token for use, so that we don't have to use `sudo` to use `/var/lib`?

Reply:
no this hardcodes `~/.kube` as the location of the config. I can change it to load from env var if it is defined.

Reply:
This config resolution is exactly the same what operator SDK does. They do not provide a way to override kubeconfig location.

So if you want the proxy to connect without copying cert and token to `/var/run/secrets/kubernetes/..` just login to k8s cluster via `oc login` it stores config to the default location in `~/.kube/config` and then proxy will use that to connect to k8s cluster.

Reply:
so operator sdk ignores `KUBECONFIG`?

I guess this is ok for now - but we must find a way to run this by setting `KUBECONFIG`, without relying on `~/.kube` and without relying on `/var/run` - there must be a way to do that

Reply:
I'm not following.  The documentation says [1] it will use `/var/....` for in cluster and the `KUBECONFIG` out of cluster.  We should be able to rely on one or the other

[1] https://github.com/kubernetes/client-go/tree/master/examples/in-cluster-client-configuration
[2]https://github.com/kubernetes/client-go/tree/master/examples/out-of-cluster-client-configuration

Reply:
@jcantrill I don't think that upstream k8s client supports `KUBECONFIG` env v ar otherwise the example would use it. The example uses a flag to set the config location https://github.com/kubernetes/client-go/blob/master/examples/out-of-cluster-client-configuration/main.go#L45

Anyways we can introduce a flag later. The current code is what operator SDK uses, we didn't have the need to override the config location in operators.